src/lib/tls/msg_server_kex.cpp

```diff
@@ -165,11 +166,18 @@ Server_Key_Exchange::Server_Key_Exchange(const std::vector<byte>& buf,
    if(kex_algo == "DH" || kex_algo == "DHE_PSK")
       {
       // 3 bigints, DH p, g, Y
-
-      for(size_t i = 0; i != 3; ++i)
-         {
-         reader.get_range<byte>(2, 1, 65535);
-         }
+      std::vector<byte> p = reader.get_range<byte>(2, 1, 65535);
+      reader.get_range<byte>(2, 1, 65535);
+      reader.get_range<byte>(2, 1, 65535);
+      
+      // protection against the LOGJAM attack
+      int key_size = p.size() * 8;
```

Comment:
Oops...

I assume this is however not needed if I leave the check in the ClientKeyExchange message?
ClientKeyExchange message processing is based on the ServerKeyExchange (or on the certificate) key, thus the check there should prevent from usage of small keys (?)